### PR TITLE
LibJS: Implement the RegExpAlloc AO

### PR DESCRIPTION
**Pull request by linusg**
Preparation for #15585. cc @leeight

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/RegExpObject.cpp [hunk 1]
```diff
@@ -156,48 +156,78 @@ void RegExpObject::initialize(Realm& realm)
...
+    // 2. The code points / or any LineTerminator occurring in the pattern shall be escaped in S as necessary to ensure that the string-concatenation of "/", S, "/", and F can be parsed (in an appropriate lexical context) as a RegularExpressionLiteral that behaves identically to the constructed regular expression. For example, if P is "/", then S could be "\/" or "\u002F", among other possibilities, but not "/", because /// followed by F would be parsed as a SingleLineComment rather than a RegularExpressionLiteral. If P is the empty String, this specification can be met by letting S be "(?:)".
+    // 3. Return S.
```
Comment (alimpfard) [inline Userland/Libraries/LibJS/Runtime/RegExpObject.cpp:230]: this seems excessively long :thinking:

File: Userland/Libraries/LibJS/Runtime/RegExpObject.cpp [hunk 2]
```diff
@@ -156,48 +156,78 @@ void RegExpObject::initialize(Realm& realm)
...
+    // 5. If F contains any code unit other than "d", "g", "i", "m", "s", "u", or "y" or if it contains the same code unit more than once, throw a SyntaxError exception.
+    auto parsed_flags_or_error = regex_flags_from_string(flags);
     if (parsed_flags_or_error.is_error())
         return vm.throw_completion<SyntaxError>(parsed_flags_or_error.release_error());
 
+    // 6. If F contains "i", let i be true; else let i be false.
+    // 7. If F contains "m", let m be true; else let m be false.
+    // 8. If F contains "s", let s be true; else let s be false.
+    // 9. If F contains "u", let u be true; else let u be false.
+    // 10. If F contains "v", let v be true; else let v be false.
```
Comment (davidot) [inline Userland/Libraries/LibJS/Runtime/RegExpObject.cpp:184]: Should these steps also be above the `auto parsed_flags_or_error = regex_flags_from_string(flags);` line?

File: Userland/Libraries/LibJS/Runtime/RegExpObject.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/RegExpObject.cpp]: Yep, broken at 120 chars.

**Review by davidot — COMMENTED** [review]
I left like 4 `no spec?` comments before getting to the second commit. Almost got worried about you.

Anyway code seems fine, adding spec comments is always nice. 
Just have one small comment. Which isn't even a problem since that method already doesn't match the spec perfectly.

File: Userland/Libraries/LibJS/Runtime/RegExpObject.cpp (hunk 2 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/RegExpObject.cpp:180]: Yep, done!

**Review by davidot — APPROVED** [review]

**Pull request merged**